Calibration/LumiAlCaRecoProducers/python/ALCARECOAlCaPCCZeroBias_cff.py

```diff
@@ -13,6 +13,10 @@
 siPixelDigisForLumiZB = siPixelDigis.clone()
 siPixelDigisForLumiZB.InputLabel = cms.InputTag("hltFEDSelectorLumiPixels")
 
+# Modify for if the phase 1 pixel detector is active
+from Configuration.Eras.Modifier_phase1Pixel_cff import phase1Pixel
+phase1Pixel.toModify(siPixelDigisForLumiZB, isUpgrade=cms.untracked.bool(True) )
```

Comment:
hi @capalmer85 @arunhep - is this parameter really untracked? 
- is this alcareco tested in one of the runTheMatrix workflows?

Reply:
Ah, good point. The necessary `SiPixelRawToDigi` parameter to be set to `True` for phase1 pixel is `UsePhase1`, which is tracked
https://github.com/cms-sw/cmssw/blob/07706592fcafef0134ff379925cc496f4858209d/EventFilter/SiPixelRawToDigi/python/SiPixelRawToDigi_cfi.py#L16-L17

AFAICT adding `isUpgrade` parameter does nothing.

Reply:
I was following this example:
https://cmssdt.cern.ch/lxr/source/DQM/SiPixelMonitorCluster/python/SiPixelMonitorCluster_cfi.py

There were several others just like this, so I followed it exactly (confirming with @davidlange6 above).  Is the point that the untracked boolean in the toModify method does nothing?  (I'm looking now.)

Reply:
@davidlange6 I believe that 1020 runs this ALCARECO on 2016 data.

Reply:
@capalmer85 The point (here) is that `SiPixelRawToDigi` does not have parameter called `isUpgrade` (while `SiPixelMonitorCluster` likely has, I didn't check). For `SiPixelRawToDigi` the phase1-specific parameter is `UsePhase1` (as you had in your first commit c15ebca90e2e13e291f88f391eafedec1e9f359f), which is a tracked parameter.

If you would have omitted the explicit type, i.e. with
```
phase1Pixel.toModify(siPixelDigisForLumiZB, isUpgrade=True)
```
you should have gotten an error message saying that "siPixelDigisForLumiZB does not have parameter isUpgrade".

(note that the point of my other comment is different, but I'd rather discuss it there)

Reply:
@makortel So you are saying that I should replace "isUpgrade" with "UsePhase1=True" as in:
https://cmssdt.cern.ch/lxr/source/EventFilter/SiPixelRawToDigi/python/SiPixelDigiToRaw_cfi.py

Is that right?

Reply:
That makes sense given that in my original PR I had:
siPixelDigisForLumiR.UsePhase1 = True

Reply:
@capalmer85 
> So you are saying that I should replace "isUpgrade" with "UsePhase1=True" as in:

Yes. But in my other comment I'm wondering why this is needed in the first place, as the proper use of `Run2_2017` era in the job configuration file should already set this parameter correctly.

Reply:
Apparently it does not in the ALCALUMIPIXEL stream.  Where is that configuration (with Run2_2017) done?  Should it be done in each ALCARECO? Once per stream?

Reply:
I have no idea about that, but I'd say that (not using `Run2_2017` era) is the real problem. Get that fixed and you should not need this PR.

Reply:
So... how does one do that?

Reply:
Let me elaborate, I have no idea how jobs for ALCALUMIPIXEL are configured. In runTheMatrix (RelVal) era is set with `--era Run2_2017` parameter to cmsDriver. In the configuration file that parameter ends up as
```
from Configuration.StandardSequences.Eras import eras
process = cms.Process('RECO',eras.Run2_2017)
```
which is what really counts.

Reply:
Ok.  I see.  So these lines are already run in the cfi from which we clone the cluster producer.  Here:
https://cmssdt.cern.ch/lxr/source/EventFilter/SiPixelRawToDigi/python/SiPixelDigiToRaw_cfi.py

In order for the "toModify" function to work, the era must be known before cloning.  So how does the configuration know the era of the data (or mc) that it is processing?  If I add the lines you suggest, then this code will not be backwards compatible, which was @davidlange6 original concern.  Do you know how/where it is set in the standard sequence for RECO?